File: lab_python_oop/Square.py
from .Rectangle import Rectangle

class Square(Rectangle):
    name = "Квадрат"

    def __init__(self, side: float, color: str):
        super().__init__(side, side, color)

    def __repr__(self) -> str:
        return ("{name}: Сторона={side}, Цвет={color}, "
                "Площадь={area:.2f}").format(
                    name=self.name(),
                    side=self.width,
                    color=str(self.color),
                    area=self.area()
                )

File: geometry_project/src/oop/square.rs
use super::{GeometricShape, Color, Rectangle};

pub struct Square {
    rectangle: Rectangle,
}

impl Square {
    pub fn new(side: f64, color: Color) -> Self {
        Self {
            rectangle: Rectangle::new(side, side, color),
        }
    }
}

impl GeometricShape for Square {
    fn area(&self) -> f64 {
        self.rectangle.area()
    }

    fn shape_name(&self) -> &'static str {
        "Квадрат"
    }
}
